unittests/util/async/AnyOperationTests.cpp

```diff
@@ -46,8 +46,7 @@ using AnyOperationDeathTest = AnyOperationTests;
 
 TEST_F(AnyOperationTests, VoidDataYieldsNoError)
 {
-    auto const noError = std::expected<impl::Any, ExecutionError>(impl::Any{});
-    EXPECT_CALL(mockOp, get()).WillOnce(Return(noError));
+    EXPECT_CALL(mockOp, get()).WillOnce(Return(std::any{}));
```

Comment:
Gcc-12 was complaining about the copy. Not quite sure why. Clang and AppleClang are fine with it.
Clangd is still complaining about a few places in the file but I'm hoping this is yet another issue that will get ironed out by moving the tooling to llvm 18 as well as better gcc-12 and clang integration in our CI docker image.